index.ts

```diff
@@ -54,8 +49,16 @@ function useLocalStorage<T>(
   useEffect(() => {
     if (typeof window === "undefined") return;
 
+    const updateLocalStorage = () => {
+      if (storedValue !== undefined) {
```

Comment:
Should we check also for `storedValue !== null`? Calling `setValue(null)` will result in the string `"null"` beign stored, which looks like the same escenario as with `undefined` to me
![image](https://user-images.githubusercontent.com/12707758/150003461-4fb403ad-cd38-4e17-852f-4e2af8292cf2.png)

Reply:
Hmmm.... so setting `null` in your view should also remove the key from local storage it seems reasonable would love to get @nsa5w take on this as well.

Reply:
@nas5w What do you think?

Reply:
Well, calling `setValue(null)` will first use _JSON.stringify_ to stringify `null` prior to storing it in `localStorage`. If you parse the stored value, you do indeed get `null` back. Try the following out in the javascript console:

```javascript
localStorage.setItem("foo", JSON.stringify(null));
null === JSON.parse(localStorage.getItem("foo"));
// true
```

So, I think `null` is a valid value to be setting in local storage.